GoogleForCreators/web-stories-wp
# Hotlinking: Do not use CORS proxy for captions and background audio

## Context



Video tracks require proper CORS headers being set, otherwise they won't load on the frontend. So our CORS proxy workaround is useless.

## Summary



This PR changes the hotlinking modal so that it fails when there is a CORS issue when trying to insert VTT files or audio files.

## Relevant Technical Choices



## To-do



## User-facing changes



No visual change initially in the dialog:
![Screenshot 2022-05-19 at 14 20 41](https://user-images.githubusercontent.com/841956/169296292-c5f62791-5b06-4c63-b2be-80a49250922c.png)

But then the new error message:

![Screenshot 2022-05-19 at 14 25 13](https://user-images.githubusercontent.com/841956/169296301-7c4c5863-0a3f-4134-9ea5-33a9e9ed5116.png)


## Testing Instructions






This PR can be tested by following these steps:

1. Try to insert a file like https://api.ardmediathek.de/subtitle-format-service/webvtt/urn:ard:subtitle:74cfd50da83320c6.vtt for background audio or video captions.
2. Verify that it works as expected and the captions are visible when viewing the story (ignore any styling differences that may appear)
3. Try to insert a file like https://wp.stories.google/stories/how-to-optimize-videos-for-web-stories/tracks/wst024_page2_my_captions.vtt for captions and see an error message about CORS


## Reviews

### Does this PR have a security-related impact?


No

### Does this PR change what data or activity we track or use?


No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #11479
